api/gen_ai/views.py

```diff
@@ -0,0 +1,62 @@
+import hmac
+import logging
+from hashlib import sha256
+
+from django.utils.crypto import constant_time_compare
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from api.gen_ai.serializers import GenAIAuthSerializer
+from codecov_auth.models import GithubAppInstallation, Owner
+from graphql_api.types.owner.owner import AI_FEATURES_GH_APP_ID
+from utils.config import get_config
+
+log = logging.getLogger(__name__)
+
+
+class GenAIAuthView(APIView):
+    permission_classes = [AllowAny]
+    serializer_class = GenAIAuthSerializer
+
+    def validate_signature(self, request):
+        key = get_config(
+            "gen_ai", "auth_secret", default=b"testixik8qdauiab1yiffydimvi72ekq"
+        )
+        if isinstance(key, str):
+            key = key.encode("utf-8")
+        expected_sig = request.headers.get("HTTP-X-GEN-AI-AUTH-SIGNATURE")
+        print(request.headers)
+        computed_sig = (
+            "sha256=" + hmac.new(key, request.body, digestmod=sha256).hexdigest()
+        )
+        if not (expected_sig and constant_time_compare(computed_sig, expected_sig)):
```

Comment:
```suggestion
        if not hmac.compare_digest(computed_sig, expected_sig):
```

I'd probably simplify this and just use `hmac`s `compare_digest`. 

I'd also suggest erroring out earlier if `expected_sig` is none with a HTTP 400. This would keep you from spending time calculating the payload signature when there's no signature in the headers anyway.

Reply:
I believe the extra `expected_sig and` is going to open this up to potential timing attacks since it does not use the constant time compare. The extra check is not needed, if it's `None` the comparison will fail.